config/DibiRow.stub

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Dibi;
+
+/**
+ * @template rowType
+ *
+ * @implements \Traversable<int, rowType>
+ * @implements \IteratorAggregate<int, rowType>
+ */
+class Row implements \Traversable, \IteratorAggregate
```

Comment:
per https://github.com/dg/dibi/blob/cab1c5b5e69f9a6c745919145b06ce82f291a7c5/src/Dibi/Row.php#L17

this should be `class Row implements \ArrayAccess, \IteratorAggregate, \Countable` (and you need to adjust the phpdocs acordingly)

Reply:
I updated the class declaration but im still lost with the phpdocs as to what should be there :(

Reply:
lookup those class definitions in [phpstan-src stubs](https://github.com/phpstan/phpstan-src/tree/1.9.x/stubs)
there you can see which class has which generics defined and which don't have generics at all.

Reply:
this is where I was looking already (i got there from https://phpstan.org/user-guide/stub-files) but this is the first stub file that I am writing :(

I updated the doc block by copying definitions from arrayaccess (https://github.com/phpstan/phpstan-src/blob/1.5.x/stubs/ArrayObject.stub) but still no luck

```php
/**
 * @template TKey of array-key
 * @template TValue
 * @implements \IteratorAggregate<TKey, TValue>
 * @implements \ArrayAccess<TKey, TValue>
 */
```

Reply:
I played arround a bit with this PR. I think stubs won't help us for the problem at hand.

just opened a discussion at https://github.com/phpstan/phpstan/discussions/8399 to brainstorm how we can get this polymorphic object represented

Reply:
Thanks, I appreciate greatly the effort :+1:

Reply:
I have no clue whether its the way to go, but you might try the suggestion here: https://github.com/phpstan/phpstan/discussions/8399#discussioncomment-4190224

Reply:
`methodReflectionExtension`, I will give it a try, thanks

Reply:
maybe its even https://phpstan.org/developing-extensions/class-reflection-extensions#properties-class-reflection-extensions